### PR TITLE
Add hybrid retrieval mode to MilvusVectorStore.

### PR DESCRIPTION
**Pull request by wxywb**
# Description

This PR add the hybrid retrieval functionality to MilvusVectorStore.

Fixes # (issue)

## New Package?

Did I fill in the `tool.llamahub` section in the `pyproject.toml` and provide a detailed README.md for my new integration or package?

- [ ] Yes
- [x] No

## Version Bump?

Did I bump the version in the `pyproject.toml` file of the package I am updating? (Except for the `llama-index-core` package)

- [x] Yes
- [ ] No

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Added new unit/integration tests
- [x] Added new notebook (that tests end-to-end)
- [ ] I stared at the code and made sure it makes sense

## Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: llama-index-integrations/vector_stores/llama-index-vector-stores-milvus/llama_index/vector_stores/milvus/utils.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    @abstractmethod
+    def encode_queries(self, queries: List[str]) -> List[Dict[int, float]]:
+        pass
+
+    @abstractmethod
+    def encode_documents(self, documents: List[str]) -> List[Dict[int, float]]:
+        pass
+
+
+class BGEM3SparseEmbeddingFunction(BaseSparseEmbeddingFunction):
```
Comment (logan-markewich) [inline llama-index-integrations/vector_stores/llama-index-vector-stores-milvus/llama_index/vector_stores/milvus/utils.py:16]: Its a little weird to call a class a function, but that might just be mean 😅

Comment (wxywb) [reply llama-index-integrations/vector_stores/llama-index-vector-stores-milvus/llama_index/vector_stores/milvus/utils.py:16]: We use the term `EmbeddingFunction` in pymilvus embedding utility, so it is consistent in context of Milvus.

**Review by logan-markewich — APPROVED** [review]

**Review by logan-markewich — APPROVED** [review]

**Comment by wxywb** [comment]
The error occurs seems the  dependency `FlagEmbedding` is not installed, but I have added it in pyproject.toml

**Comment by eljefedelrodeodeljefe** [comment]
@wxywb I believe this change causes `llama-index-vector-stores-milvus` to install torch and nvidia-* to be installed, which is a 3GB weight. In my case this causes installation issues on GH actions. Is there no way of optionally importing FlagEmbedding?

**Comment by wxywb** [comment]
> @wxywb I believe this change causes `llama-index-vector-stores-milvus` to install torch and nvidia-*, which is a 3GB+ weight. In my case this causes installation issues on GH actions. Is there no way of optionally importing FlagEmbedding?

i understand this requirement, i am going to remove this dependency.

**Comment by wxywb** [comment]
@eljefedelrodeodeljefe 
I have just removed  FlagEmbedding as dependency, thanks for pointing out.
https://github.com/run-llama/llama_index/pull/13767

**Pull request merged**